usaspending_api/agency/v2/views/obligations_by_award_category.py

```diff
@@ -0,0 +1,76 @@
+from rest_framework.request import Request
+from rest_framework.response import Response
+from typing import Any
+
+from usaspending_api.agency.v2.views.agency_base import AgencyBase
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.elasticsearch.filter_helpers import create_fiscal_year_filter
+from usaspending_api.common.elasticsearch.search_wrappers import TransactionSearch
+from usaspending_api.common.query_with_filters import QueryWithFilters
+
+
+class ObligationsByAwardCategory(AgencyBase):
+    """
+    Returns a breakdown of obligations by award category (contracts, IDVs, grants, loans, direct payments, other)
+    within the requested fiscal year (or current FY).
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/toptier_code/obligations_by_award_category.md"
+
+    @cache_response()
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+
+        results = self.query_elasticsearch()
+
+        formatted_results = self.format_results(results)
+
+        return Response(formatted_results)
+
+    def query_elasticsearch(self):
+        filters = {}
+        filters["time_period"] = create_fiscal_year_filter(self.fiscal_year)
+        filters["agencies"] = [{"toptier_id": self.agency_id, "type": "awarding", "tier": "toptier"}]
+
+        filter_query = QueryWithFilters.generate_transactions_elasticsearch_query(filters)
+        search = TransactionSearch().filter(filter_query)
+
+        search.aggs.bucket("group_by_award_category", "terms", field="award_category").metric(
+            "obligations", "sum", field="generated_pragmatic_obligation"
+        )
+
+        return search.handle_execute().to_dict()
+
+    def format_results(self, results):
+        results_map = {}
+        formatted_categories = []
+        total = 0.0
+        other_total = 0.0
+
+        category_list = ["contract", "direct_payment", "grant", "idv", "loan", "other"]
+
+        # Reformat List of Objects as Map, so they can be alphabetized later
+        for obj in results["aggregations"]["group_by_award_category"]["buckets"]:
+            key = obj["key"]
+            obligation_value = obj["obligations"]["value"]
+            total += obligation_value
+
+            if key in category_list:
```

Comment:
Please double check the names for the keys and the category list - I believe some direct payments and loans are getting sent into the `other` category since the keys are `direct payment` and `loans` not `direct_payment` and `loan`

Reply:
Good call. Fixed those two. The others looked okay